server/boot/user.js

```diff
@@ -608,7 +608,14 @@ module.exports = function(app) {
     return User.findById(req.accessToken.userId, function(err, user) {
       if (err) { return next(err); }
       return user.updateAttribute('password', password, function(err) {
-        if (err) { return next(err); }
+        if (err) {
```

Comment:
Ah I see. Unfortunately db connection errors will also be reported here. There must be some way of distinguishing these two errors?